### PR TITLE
ARROW-12741: [CI] Configure Crossbow GitHub Token for Nightly Builds

### PR DESCRIPTION
**Pull request by kszucs**
This partially reverts commit dae3fcccca9bba114913b2b09564127bc0ee779e.

### CODE REVIEW

File: dev/archery/archery/cli.py [hunk 1]
```diff
@@ -789,7 +789,9 @@ def integration(with_all=False, random_seed=12345, **args):
               default='-', required=True)
 @click.option('--arrow-token', envvar='ARROW_GITHUB_TOKEN',
               help='OAuth token for responding comment in the arrow repo')
-def trigger_bot(event_name, event_payload, arrow_token):
+@click.option('--crossbow-token', '-ct', envvar='CROSSBOW_GITHUB_TOKEN',
+              help='OAuth token for pushing to the crossow repository')
+def trigger_bot(event_name, event_payload, arrow_token, crossbow_token):
```
Comment (kou) [inline dev/archery/archery/cli.py:794]: We don't need to revert this change because `crossbow_token` isn't used in this function.

File: dev/tasks/macros.jinja [hunk 2]
```diff
@@ -87,7 +87,7 @@ on:
       {% endfor %}
     {% endif %}
     env:
-      CROSSBOW_GITHUB_TOKEN: {{ '${{ secrets.GITHUB_TOKEN }}' }}
+      CROSSBOW_GITHUB_TOKEN: {{ '${{ secrets.CROSSBOW_GITHUB_TOKEN }}' }}
```
Comment (kou) [inline dev/tasks/macros.jinja:90]: Can we try the following?

```suggestion
      CROSSBOW_GITHUB_TOKEN: {{ '${{ secrets.CROSSBOW_GITHUB_TOKEN || secrets.GITHUB_TOKEN }}' }}
```

Comment (kszucs) [reply dev/tasks/macros.jinja:90]: LGTM

**Comment by kszucs** [comment]
Submitted build https://github.com/ursacomputing/crossbow/branches/all?query=build-300

Using: `archery crossbow submit wheel-manylinux2010-cp36-amd64 --arrow-branch master`

**Review by kszucs — COMMENTED** [review]
+1
